app/scripts/lib/snap-keyring/snap-keyring.ts

```diff
@@ -51,6 +56,29 @@ export const snapKeyringBuilder = (
         const addresses = await getKeyringController().getAccounts();
         return addresses.includes(address.toLowerCase());
       },
+      redirectUser: async (snapId: string, url: string, message: string) => {
+        // Either url or message must be defined
+        if (url.length > 0 || message.length > 0) {
+          const isBlocked = await isBlockedUrl(url, getPhishingController());
```

Comment:
What is the reasoning for showing the popup if the URL is blocked anyway? 🤔

Reply:
It is another opportunity for us to inform the user of potential malicious activity.

Reply:
Okay, one nit then, should we disable showing the "follow the instructions..." copy then. That doesn't seem useful if the instructions are malicious?

Reply:
I made the update in the latest change
<img width="425" alt="Screenshot 2023-10-12 at 10 48 01 AM" src="https://github.com/MetaMask/metamask-extension/assets/22918444/1d70c225-6486-43a1-b351-ab7a14eb8b56">

Reply:
@FrederikBolding I also changed the cancel button to say close. This matches the copy in [these](https://www.figma.com/file/K0csZegZ5XQ6TjfKk3n4Z8/Add-dialog-to-inform-about-redirection-to-snap-site-%2335?type=design&node-id=604-2586&mode=design&t=XHzW2Gm7AXUtIUhL-0) designs and makes more sense